base_report_to_printer/ir_report.py

```diff
@@ -118,6 +143,19 @@ def behaviour(self, cr, uid, ids, context=None):
             if report.printing_printer_id:
                 printer = report.printing_printer_id
 
+            # Retrieve report-company specific values
+            act_ids = printing_act_obj.search(
+                cr, uid,
+                [('report_id', '=', report.id),
+                 ('company_id', '=', company_id),
+                 ('action', '!=', 'user_default')], context=context)
+            if act_ids:
+                company_action = printing_act_obj.behaviour(
+                    cr, uid, act_ids[0], context)
+                action = company_action['action']
+                if company_action['printer']:
+                    printer = company_action['printer']
+
             # Retrieve report-user specific values
             act_ids = printing_act_obj.search(
                 cr, uid,
```

Comment:
Since we have a company field now, maybe it could be nice to also look for a printer for a user in a company in the search domain below (for multi-company users).

I mean a query like the following (for : 
```
[('report_id', '=', report.id),
                  ('user_id', '=', uid),
                  ('action', '!=', 'user_default'),
                  '|', ('company_id', '=', company_id),
                   ('company_id', '=', False),
]
```